corehq/apps/enterprise/interface.py

```diff
@@ -0,0 +1,194 @@
+                },
+                {
+                    'name': GatewayTypeFilter.slug,
+                    'value': GatewayTypeFilter.get_value(self.request, self.domain)
+                },
+        ]
+
+    @property
+    def get_all_rows(self):
+        query = self.sms_billables
+        query = query.order_by(self.sort_field)
+        return self._format_billables(query)
+
+    @property
+    def total_records(self):
+        query = self.sms_billables
+        return query.count()
+
+    @property
+    def rows(self):
+        query = self.sms_billables
+        query = query.order_by(self.sort_field)
+
+        sms_billables = query[self.pagination.start:(self.pagination.start + self.pagination.count)]
+        return self._format_billables(sms_billables)
+
+    def _format_billables(self, sms_billables):
+        return [
+            [
+                sms_billable.date_sent,
+                sms_billable.domain,
+                {
+                    INCOMING: _("Incoming"),
+                    OUTGOING: _("Outgoing"),
+                }[sms_billable.direction],
+                sms_billable.multipart_count,
+                sms_billable.gateway_fee.criteria.backend_api_id if sms_billable.gateway_fee else "",
+                sms_billable.gateway_charge,
+                sms_billable.usage_charge,
+                sms_billable.gateway_charge + sms_billable.usage_charge,
+                sms_billable.log_id,
+                sms_billable.is_valid,
+                sms_billable.date_created,
+            ]
+            for sms_billable in sms_billables
+        ]
+
+    @property
+    def sms_billables(self):
+        datespan = DateSpan(DateSentFilter.get_start_date(self.request), DateSentFilter.get_end_date(self.request))
+        selected_billables = SmsBillable.get_billables_sent_between(datespan)
+        if DateCreatedFilter.use_filter(self.request):
+            date_span = DateSpan(
+                DateCreatedFilter.get_start_date(self.request), DateCreatedFilter.get_end_date(self.request)
+            )
+            selected_billables = SmsBillable.filter_selected_billables_by_date(selected_billables, date_span)
+        show_billables = ShowBillablesFilter.get_value(
+            self.request, self.domain)
+        if show_billables:
+            selected_billables = SmsBillable.filter_selected_billables_show_billables(
+                selected_billables, show_billables
+            )
+        domain = EnterpriseDomainFilter.get_value(self.request, self.domain)
+        if domain:
+            selected_billables = selected_billables.filter(
+                domain=domain,
+            )
+        else:
+            account = BillingAccount.get_account_by_domain(self.request.domain)
+            domains = Subscription.get_active_domains_for_account(account)
+            selected_billables = selected_billables.filter(
+                domain__in=domains
+            )
```

Comment:
There's some unfortunate duplication here, as both the filter and the else clause have to do the same logic to find out the allowed domains. It looks like we could get around this by instantiating the `EnterpriseDomainFilter` rather than using the class object, and then calling a common method to grab the allowed domains (right now calling `.options()` would require peeling off the first element in each tuple. It would likely be better to have `options()` call a helper method that we could call directly in the else clause). Another option would be to create a helper method outside of the filter, and calling that within options and the else clause